pkg/collector/corechecks/servicediscovery/impl_linux.go

```diff
@@ -25,40 +29,58 @@ func init() {
 }
 
 type linuxImpl struct {
-	getSysProbeClient processnet.SysProbeUtilGetter
-	time              timer
+	getDiscoveryServices func(client *http.Client) (*model.ServicesResponse, error)
+	time                 timer
 
 	ignoreCfg         map[string]bool
 	containerProvider proccontainers.ContainerProvider
 
 	ignoreProcs       map[int]bool
 	aliveServices     map[int]*serviceInfo
 	potentialServices map[int]*serviceInfo
+
+	sysProbeClient *http.Client
 }
 
 func newLinuxImpl(ignoreCfg map[string]bool, containerProvider proccontainers.ContainerProvider) (osImpl, error) {
 	return &linuxImpl{
-		getSysProbeClient: processnet.GetRemoteSystemProbeUtil,
-		time:              realTime{},
-		ignoreCfg:         ignoreCfg,
-		containerProvider: containerProvider,
-		ignoreProcs:       make(map[int]bool),
-		aliveServices:     make(map[int]*serviceInfo),
-		potentialServices: make(map[int]*serviceInfo),
+		getDiscoveryServices: getDiscoveryServices,
+		time:                 realTime{},
+		ignoreCfg:            ignoreCfg,
+		containerProvider:    containerProvider,
+		ignoreProcs:          make(map[int]bool),
+		aliveServices:        make(map[int]*serviceInfo),
+		potentialServices:    make(map[int]*serviceInfo),
+		sysProbeClient:       sysprobeclient.Get(pkgconfigsetup.SystemProbe().GetString("system_probe_config.sysprobe_socket")),
 	}, nil
 }
 
-func (li *linuxImpl) DiscoverServices() (*discoveredServices, error) {
-	socket := pkgconfigsetup.SystemProbe().GetString("system_probe_config.sysprobe_socket")
-	sysProbe, err := li.getSysProbeClient(socket)
+func getDiscoveryServices(client *http.Client) (*model.ServicesResponse, error) {
+	url := sysprobeclient.ModuleURL(sysconfig.DiscoveryModule, "/services")
+	req, err := http.NewRequest(http.MethodGet, url, nil)
 	if err != nil {
-		return nil, errWithCode{
-			err:  err,
-			code: errorCodeSystemProbeConn,
-		}
+		return nil, err
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("got non-success status code: url: %s, status_code: %d", req.URL, resp.StatusCode)
 	}
 
-	response, err := sysProbe.GetDiscoveryServices()
+	res := &model.ServicesResponse{}
+	if err := json.NewDecoder(resp.Body).Decode(res); err != nil {
+		return nil, err
+	}
+	return res, nil
+}
+
+func (li *linuxImpl) DiscoverServices() (*discoveredServices, error) {
+	response, err := li.getDiscoveryServices(li.sysProbeClient)
```

Comment:
the original implementation (systemprobeclient) had retry mechanism, here we don't have that